File: 프로그래머스/lv2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var answer = 0;

    people.sort((a, b) => b - a);
    for (let i = 0, j = people.length - 1; i <= j; ++i) {
        if (limit >= people[i] + people[j]) {
            ++answer;
            --j;
        } else ++answer;
    }

    return answer;
}

File: 프로그래머스/lv2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    int leftPointer = 0, rightPointer = people.size() - 1;

    sort(people.begin(), people.end());
    
    while (leftPointer <= rightPointer)
    {
        if (people[leftPointer] + people[rightPointer] <= limit)
            ++leftPointer;

        --rightPointer;
        ++answer;
    }
    return answer;
}
